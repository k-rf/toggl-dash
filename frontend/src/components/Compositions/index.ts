export * from "./Form";
export * from "./RhfTextField";
