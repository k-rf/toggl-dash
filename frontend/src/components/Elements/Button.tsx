import { Button as MuiButton, ButtonProps as MuiButtonProps } from "@mui/material";

type Props = Pick<
  MuiButtonProps,
  "children" | "color" | "disabled" | "fullWidth" | "onClick" | "variant"
>;

export const Button = (props: Props) => {
  return <MuiButton {...props} />;
};
