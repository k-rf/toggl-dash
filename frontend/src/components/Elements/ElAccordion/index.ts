export * from "./ElAccordion";
export * from "./ElAccordionDetails";
export * from "./ElAccordionSummary";
