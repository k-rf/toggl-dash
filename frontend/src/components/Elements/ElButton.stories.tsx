import { Box, Stack } from "@mui/material";
import type { Meta, StoryObj } from "@storybook/react";

import { ElButton, ElButtonProps } from "./ElButton";

export default { component: ElButton } as Meta<ElButtonProps>;

type Story = StoryObj<ElButtonProps>;

export const Default: Story = {
  args: {
    children: "Button",
  },
  render: (args) => {
    return <Buttons {...args} />;
  },
};

export const Colors: Story = {
  render: (args) => (
    <Stack direction="column" spacing={2}>
      <Buttons {...args} color="primary">
        Primary
      </Buttons>
      <Buttons {...args} color="secondary">
        Secondary
      </Buttons>
      <Buttons {...args} color="success">
        Success
      </Buttons>
      <Buttons {...args} color="info">
        Info
      </Buttons>
      <Buttons {...args} color="warning">
        Warning
      </Buttons>
      <Buttons {...args} color="error">
        Error
      </Buttons>
      <Buttons {...args} color="inherit">
        Inherit
      </Buttons>
    </Stack>
  ),
};

const Buttons = (args: ElButtonProps) => {
  return (
    <Stack direction="row" spacing={2}>
      <Box width={160}>
        <ElButton {...args} variant="contained" fullWidth />
      </Box>
      <Box width={160}>
        <ElButton {...args} variant="outlined" fullWidth />
      </Box>
      <Box width={160}>
        <ElButton {...args} variant="text" fullWidth />
      </Box>
    </Stack>
  );
};
