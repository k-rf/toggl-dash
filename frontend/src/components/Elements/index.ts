export * from "./ElButton";
export * from "./ElPaper";
export * from "./ElTextField";
export * from "./ElAccordion";
