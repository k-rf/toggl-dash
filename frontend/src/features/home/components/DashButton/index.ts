export * from "./DashButton";
export * from "./DashButtonDetailListItem";
