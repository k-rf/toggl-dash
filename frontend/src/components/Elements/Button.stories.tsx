import { Box, Stack } from "@mui/material";
import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import { Button } from "./Button";

type Meta = ComponentMeta<typeof Button>;
type Story = ComponentStoryObj<typeof Button>;

export default { title: "Elements/Button", component: Button } as Meta;

export const Default: Story = {
  args: {
    children: "ボタン",
    color: "primary",
  },
};

export const Variant: Story = {
  ...Default,
  args: {
    ...Default.args,
  },
  render: (args) => {
    return <Buttons {...args} />;
  },
};

export const Disabled: Story = {
  ...Default,
  args: {
    ...Default.args,
    disabled: true,
  },
  render: (args) => {
    return <Buttons {...args} />;
  },
};

export const Colors: Story = {
  ...Default,
  args: {
    ...Default.args,
    fullWidth: true,
  },
  render: (args) => {
    return (
      <Stack direction={"column"} spacing={2}>
        <Buttons {...args} color="primary" width={120}>
          primary
        </Buttons>
        <Buttons {...args} color="secondary" width={120}>
          secondary
        </Buttons>
        <Buttons {...args} color="success" width={120}>
          success
        </Buttons>
        <Buttons {...args} color="warning" width={120}>
          warning
        </Buttons>
        <Buttons {...args} color="error" width={120}>
          error
        </Buttons>
        <Buttons {...args} color="info" width={120}>
          info
        </Buttons>
      </Stack>
    );
  },
};

const Buttons = (args: Story["args"] & { width?: number }) => {
  const { width, ...props } = args;
  return (
    <Stack direction={"row"} spacing={2}>
      <Box width={width}>
        <Button {...props} variant="contained" />
      </Box>
      <Box width={width}>
        <Button {...props} variant="outlined" />
      </Box>
      <Box width={width}>
        <Button {...props} variant="text" />
      </Box>
    </Stack>
  );
};
