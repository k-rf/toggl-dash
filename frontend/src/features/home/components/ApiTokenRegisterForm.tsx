import { InputAdornment, Stack } from "@mui/material";
import { KeyboardEventHandler, MouseEventHandler, useState } from "react";

import { Form, HandleSubmit, Methods } from "~/components/Compositions";
import { ElButton, ElTextField } from "~/components/Elements";
import { ElIconButton } from "~/components/Elements/ElIconButton";
import { ElTypography } from "~/components/Elements/ElTypography";
import { VisibilityIcon, VisibilityOffIcon } from "~/components/Icons";
import { z } from "~/lib/zod";

const schema = z.object({
  apiToken: z.string().min(1),
});

type ApiTokenValues = z.infer<typeof schema>;

type Props = { onSubmit?: (value: ApiTokenValues) => Promise<void> | void };

export const ApiTokenRegisterForm = (props: Props) => {
  const [visible, setVisible] = useState(false);

  const submit = (methods: Methods<ApiTokenValues>) => {
    return methods.handleSubmit(async (value) => {
      methods.reset({ apiToken: "" });

      await props.onSubmit?.(value);
    });
  };

  const handleSubmit: HandleSubmit<ApiTokenValues, MouseEventHandler> = (methods) => {
    return submit(methods);
  };

  const handleKeyDown: HandleSubmit<ApiTokenValues, KeyboardEventHandler> = (methods) => {
    return (e) => {
      if (e.code === "Enter") {
        e.preventDefault();

        if (e.ctrlKey) {
          return submit(methods)(e);
        }
      }
    };
  };

  const handleShow = () => {
    setVisible(true);
  };

  const handleMask = () => {
    setVisible(false);
  };

  return (
    <Form<ApiTokenValues> options={{ mode: "all" }} schema={schema}>
      {(methods) => {
        const { error } = methods.getFieldState("apiToken");
        const {
          errors: { apiToken },
          isValid,
        } = methods.formState;

        return (
          <Stack spacing={1}>
            <ElTypography>Toggl の API トークンを入力してください</ElTypography>
            <ElTextField
              fullWidth
              placeholder="API トークン"
              type={visible ? "text" : "password"}
              error={Boolean(error)}
              onKeyDown={handleKeyDown(methods)}
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    <ElIconButton
                      disableTouchRipple
                      onMouseDown={handleShow}
                      onMouseUp={handleMask}
                      onMouseLeave={handleMask}
                      data-testid="api-token-register-form-visibility-toggle-icon-button"
                    >
                      {visible ? <VisibilityOffIcon /> : <VisibilityIcon />}
                    </ElIconButton>
                  </InputAdornment>
                ),
              }}
              registration={methods.register("apiToken")}
            />
            <ElButton
              disabled={!isValid}
              onClick={handleSubmit(methods)}
              data-testid="api-token-register-form-submit-button"
            >
              保存する
            </ElButton>
            <ElTypography variant="body2" color="error">
              {apiToken?.message}
            </ElTypography>
          </Stack>
        );
      }}
    </Form>
  );
};

export type { Props as ApiTokenRegisterFormProps };
