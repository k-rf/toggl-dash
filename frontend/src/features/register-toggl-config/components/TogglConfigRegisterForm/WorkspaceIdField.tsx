import { KeyboardEventHandler } from "react";
import { FieldError, UseFormRegisterReturn } from "react-hook-form";

import { ElTextField } from "~/components/Elements";

type Props = {
  error?: FieldError;
  onKeyDown: KeyboardEventHandler<Element>;
  registration: Partial<UseFormRegisterReturn>;
};

export const WorkspaceIdField = (props: Props) => {
  return (
    <ElTextField
      id="workspace-id-field"
      fullWidth
      size="small"
      placeholder="ワークスペース ID"
      type="text"
      error={Boolean(props.error)}
      onKeyDown={props.onKeyDown}
      registration={props.registration}
    />
  );
};
