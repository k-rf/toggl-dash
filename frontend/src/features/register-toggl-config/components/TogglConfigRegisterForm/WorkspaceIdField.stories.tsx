import type { Meta, StoryObj } from "@storybook/react";

import { WorkspaceIdField } from "./WorkspaceIdField";

export default { component: WorkspaceIdField } as Meta<typeof WorkspaceIdField>;

type Story = StoryObj<typeof WorkspaceIdField>;

export const Default: Story = {};
