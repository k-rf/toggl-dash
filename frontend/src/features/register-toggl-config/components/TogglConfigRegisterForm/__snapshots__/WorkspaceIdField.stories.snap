// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features/Register-toggl-config/Components/TogglConfigRegisterForm/WorkspaceIdField > Default 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="workspace-id-field"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="ワークスペース ID"
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
